sound/soc/intel/boards/sof_sdw.c

```diff
@@ -253,12 +253,21 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 	},
 };
 
-static inline int find_codec_info_part(unsigned int part_id)
+static inline int find_codec_info_part(u64 adr)
 {
+	unsigned int part_id, sdw_version;
 	int i;
 
+	part_id = SDW_PART_ID(adr);
+	sdw_version = SDW_VERSION(adr);
 	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++)
-		if (part_id == codec_info_list[i].id)
+		/*
+		 * A codec info is for all sdw version with the part id if
+		 * v_id is not specified in the codec info.
+		 */
+		if (part_id == codec_info_list[i].id &&
+		    (!codec_info_list[i].v_id ||
```

Comment:
do we need this test? The version start at one, it's never zero for all SoundWire devices, and on our side we can update the tables as needed.

Reply:
while we're changing can we also slightly modify the function to return i when the match is found and return -EINVAL outside?

Reply:
I don't get your comment @ranj063, we already return i and -EINVAL?

Reply:
@plbossart that test is exactly for the reason not to have to update all entries of `codec_info_list` immediately. It's a kind of a "backward compatibility" mode - entries without `.v_id` explicitly set will still work, and those, where it's set, since it cannot be 0 as you say, the code will check the version.

Reply:
> @plbossart that test is exactly for the reason not to have to update all entries of `codec_info_list` immediately. It's a kind of a "backward compatibility" mode - entries without `.v_id` explicitly set will still work, and those, where it's set, since it cannot be 0 as you say, the code will check the version.

Thanks @lyakh and the other reason is that some codecs with different version id can use the same codec info. For example, we can use the existing  rt715 codec info for rt715 sdca version codec. So a codec info will be used for all codecs with the same part id if version_id is not specified.